usr/lib/cca_stdll/cca_specific.c

```diff
@@ -238,6 +238,27 @@ static const MECH_LIST_ELEMENT cca_mech_list[] = {
 static const CK_ULONG cca_mech_list_len =
                         (sizeof(cca_mech_list) / sizeof(MECH_LIST_ELEMENT));
 
+
+/* Helper function: build attribute and update template */
+static CK_RV build_update_attribute(TEMPLATE * tmpl,
+                                    CK_ATTRIBUTE_TYPE type,
+                                    CK_BYTE * data, CK_ULONG data_len)
```

Comment:
We could move this function to common code (e.g. utility.c, were build_attribute() is already) so that others could also utilize it, since many other places also have the sequence of build_attribute() followed by template_update_attribute(), but that's not something that should hold off this PR.

Reply:
yes, this could and should be done. And - as this function avoids the free() if update_attribute() fails, all the places where build_attribute() + update_attribute() is used would benefit from this.